### ISSUE TITLE
Fallback environments when looking up SLS

### ISSUE THREAD

**Issue by eliasp** [saltstack/salt#21394]
**The problem I'm trying to solve:**
My GitFS based fileserver uses 8 git repositories.
When developing a new feature, bugfix or something else, I usually create a new branch/environment like `dev-feature-linux_domain_membership` or `dev-bug-wrong_ntp_packages` in the affected repository.

But when testing now these changes during development, I end up with a Minion not finding a lot of SLS because all other repositories might just have a `master`/`base` branch/environment:

```
Specified SLS ntp in saltenv dev-feature-linux_domain_membership is not available on the salt master
```

In this case, `ntp` is formula in my `salt-formulas.git` repository which just consist of git submodules of public formulas, so there's no `dev-feature-linux_domain_membership` branch/environment there.

**How I work around it:**
As a workaround I end up creating the same branch/environment in all other repositories.

This has several drawbacks:
- I need to keep the workaround-branches in the other repos constantly in sync with their master
- It requires a lot of manual effort
- It is error prone

**How I would like to see this solved:**
Make it possible either per environment or globally to define a fallback environment (by default `base`) which is used in case a SLS can't be found.

**Comment by eliasp** [comment]
Sorry for the noise - some GitFS related trouble I had here lead me to the conclusion stated above, but it seems more likely to be caused by issues with symlinks in git repositories. See also: #21413

**Comment by eliasp** [comment]
Ok, after working around #21413 it turns out this one is still valid - reopening.

**Comment by rallytime** [comment]
Thanks @eliasp - can you post your versions report, please, just to keep everything straight?

ping @terminalmage

**Comment by eliasp** [comment]
This feature request is based on the feature set as available in `2014.7.1`:
```
               Salt: 2014.7.1
             Python: 2.7.6 (default, Mar 22 2014, 22:59:56)
             Jinja2: 2.7.2
           M2Crypto: 0.21.1
     msgpack-python: 0.3.0
       msgpack-pure: Not Installed
           pycrypto: 2.6.1
            libnacl: Not Installed
             PyYAML: 3.10
              ioflo: Not Installed
              PyZMQ: 14.0.1
               RAET: Not Installed
                ZMQ: 4.0.4
               Mako: 0.9.1
```

**Comment by The-Loeki** [comment]
+1 on this one.
Maybe do something like `saltenv=dev,test,base` with envs finding & matching in order
